pkg/proto/nbcontract/v1/config.proto

```diff
@@ -227,4 +246,35 @@ message NetworkConfig {
   string cni_plugins_url = 7;
   string subnet = 8;
   string route_table = 9;
+}
+
+message CustomSearchDomain {
+  string custom_search_domain_name = 1;
+  string custom_search_domain_realm_user = 2;
+  string custom_search_domain_realm_password = 3;
+}
+
+message CustomLinuxOSConfig {
+  //this should change, we should not have any should fields, either modify scripts or calculate them on VHD
+  bool should_config_swap_file = 1;
+  bool should_config_transparent_hugepage = 2;
+  bool should_config_containerd_ulimits = 3;
+  string containerd_ulimits = 4;
+  string swap_file_size_mb = 5;
+  string transparent_hugepage = 6;
+  string transparent_defrag = 7;
+}
+
+message TLSBootstrapping {
+  // discuss whether this should be featureState or bool
+  // understand how defaulting works for enum, because we want default to be false
+  FeatureState tls_bootstrapping = 1;
```

Comment:
should prob also add the override for secure TLS bootstrap AAD resource

Reply:
ack, this right now is for testing only, will revisit this!